zookeeper-server/zookeeper-server-3.5.6/src/main/java/com/yahoo/vespa/zookeeper/Reconfigurer.java

```diff
@@ -86,7 +88,26 @@ private void reconfigure(ZookeeperServerConfig newConfig) {
         joiningServers = joiningServers.isEmpty() ? null : joiningServers;
         log.log(Level.INFO, "Will reconfigure ZooKeeper cluster. Joining servers: " + joiningServers +
                             ", leaving servers: " + leavingServers);
-        zooKeeperReconfigure(connectionSpec(activeConfig), joiningServers, leavingServers);
+
+        String connectionSpec = connectionSpec(activeConfig);
+        boolean reconfigured = false;
+        Instant end = Instant.now().plus(retryReconfigurationPeriod);
+        // Loop reconfiguring since we might need to wait until another reconfiguration is finished before we can succeed
+        while ( ! reconfigured && Instant.now().isBefore(end)) {
+            try {
+                zooKeeperReconfigure(connectionSpec, joiningServers, leavingServers);
+                reconfigured = true;
```

Comment:
Consider logging how long reconfig took, so that we fine-adjust the timeouts.

Reply:
Yes, good idea